app/javascript/components/pages/SearchEdit.js

```diff
@@ -1,13 +1,392 @@
-import React from "react"
+import React, { Component } from 'react'
+import {
+  Button,
+  Col,
+  Form,
+  FormGroup,
+  Input,
+  Label,
+  Row
+} from 'reactstrap'
+import { Redirect } from 'react-router-dom'
 import PropTypes from "prop-types"
+import cityData from "../TicityCities.js"
+import findCities from "../../helpers/find-cities"
+
 class SearchEdit extends React.Component {
-    render() {
-        return (
-            <React.Fragment>
-                <h3>Edit Search Page</h3>
-            </React.Fragment>
-        );
-    }
+  constructor(props) {
+      super(props)
+      searchKey: ""
+      searchValue: 0
+      this.state = {
+          form: {
+              city1: this.props.search.city1,
+              city2: this.props.search.city2,
+              city3: this.props.search.city3,
+              housing: this.props.search.housing,
+              costofliving: this.props.search.costofliving,
+              startups: this.props.search.startups,
+              capital: this.props.search.capital,
+              travel: this.props.search.travel,
+              commute: this.props.search.commute,
+              business: this.props.search.business,
+              safety: this.props.search.safety,
+              healthcare: this.props.search.healthcare,
+              education: this.props.search.education,
+              environment: this.props.search.environment,
+              economy: this.props.search.economy,
+              taxation: this.props.search.taxation,
+              internetaccess: this.props.search.internetaccess,
+              leisure: this.props.search.leisure,
+              tolerance: this.props.search.tolerance,
+              outdoors: this.props.search.outdoors,
+              user_id: this.props.current_user.id,
+          },
+          success: false
+      }
+  }
+
+  findCities = (cityData) => {
+      let { form } = this.state
+      const newForm = findCities(cityData, form, this.searchKey, this.searchValue)
+      this.setState({ form: newForm })
+  }
+
+  handleChange = (e) => {
+      let { form } = this.state
+      this.searchKey = e.target.name
+      this.searchValue = e.target.value
+      form[e.target.name] = e.target.value
+      this.setState({ form: form })
+  }
+
+  handleSubmit = (e) => {
+      e.preventDefault()
+      this.findCities(cityData)
+      this.props.updateSearch(this.state.form, this.props.search.id)
+      this.setState({ success: true })
+  }
+
+  render() {
+    return (
+      <>
+        <h3>Edit Categories</h3>
+        <div className="body-container">
+            <div className="form">
+              <Form>
+                <Row>
+                  <Col>
+                    <FormGroup>
+                      <Row>
+                        <Col>
+                            <Label>Housing</Label>
```

Comment:
I think it might be worth it to define all these categories in a separate file, and then just map over those here, rather than writing them all out in JSX. Something like: 
```
[
  { label: "Housing", name: "housing"},
...moreCategories
]
```

Reply:
if you dont mind, we would like to take this feedback honestly, but icebox it for now so that we can proceed and make the update later